src/plugins/query_enhancements/common/utils.ts

```diff
@@ -143,31 +59,37 @@ export const fetch = (context: EnhancedFetchContext, query: Query, aggConfig?: Q
       body,
       signal,
     })
-  ).pipe(tap(handleDataFrameError));
+  ).pipe(tap(handleFetchError));
 };
 
-export const fetchDataFrame = (context: EnhancedFetchContext, query: Query, df: IDataFrame) => {
-  const { http, path, signal } = context;
-  const body = JSON.stringify({ query: { ...query, format: 'jdbc' }, df });
-  return from(
-    http.fetch({
-      method: 'POST',
-      path,
-      body,
-      signal,
-    })
-  ).pipe(tap(handleDataFrameError));
+export const handleQueryStatusPolling = <T, P = void>(
+  fetchQueryStatus: FetchFunction<T, P>,
+  interval: number = 5000
+): Promise<T> => {
+  return new Promise((resolve, reject) => {
+    const pollQueryStatus = async () => {
+      try {
+        const response: any = await fetchQueryStatus();
+
+        if (response.status === 'SUCCESS') {
+          resolve(response);
+        } else if (response.status === 'FAILED') {
+          reject(new Error('Job failed'));
+        } else {
+          setTimeout(pollQueryStatus, interval);
+        }
+      } catch (error) {
+        reject(error);
+      }
+    };
+
+    pollQueryStatus();
+  });
 };
 
-export const fetchDataFramePolling = (context: EnhancedFetchContext, df: IDataFrame) => {
-  const { http, path, signal } = context;
-  const queryId = df.meta?.queryId;
-  const dataSourceId = df.meta?.queryConfig?.dataSourceId;
-  return from(
-    http.fetch({
-      method: 'GET',
-      path: `${path}/${queryId}${dataSourceId ? `/${dataSourceId}` : ''}`,
-      signal,
-    })
-  );
+export const buildQueryStatusConfig = (response: any) => {
+  return {
+    queryId: response.data.queryId,
+    sessionId: response.data.sessionId,
```

Comment:
Can we please make sure the sessionId is stored per - MDS cluster and flint datasource?